command/hcl2_upgrade_test.go

```diff
@@ -19,22 +19,22 @@ func Test_hcl2_upgrade(t *testing.T) {
 		exitCode  int
 		exitEarly bool
 	}{
-		{folder: "unknown_builder", flags: []string{}, exitCode: 1},
-		{folder: "complete", flags: []string{"-with-annotations"}},
-		{folder: "without-annotations", flags: []string{}},
-		{folder: "minimal", flags: []string{"-with-annotations"}},
-		{folder: "source-name", flags: []string{"-with-annotations"}},
-		{folder: "error-cleanup-provisioner", flags: []string{"-with-annotations"}},
-		{folder: "aws-access-config", flags: []string{}},
-		{folder: "escaping", flags: []string{}},
-		{folder: "vsphere_linux_options_and_network_interface", exitCode: 1, flags: []string{}},
+		{folder: "unknown_builder", flags: []string{}, exitCode: 1}, // warn for unknown components not tracked in knownPluginPrefixes
+		{folder: "complete", flags: []string{"-with-annotations"}, exitCode: 0},
+		{folder: "without-annotations", flags: []string{}, exitCode: 0},
+		{folder: "minimal", flags: []string{"-with-annotations"}, exitCode: 0},
+		{folder: "source-name", flags: []string{"-with-annotations"}, exitCode: 0},
+		{folder: "error-cleanup-provisioner", flags: []string{"-with-annotations"}, exitCode: 0},
+		{folder: "aws-access-config", flags: []string{}, exitCode: 0},
+		{folder: "escaping", flags: []string{}, exitCode: 0},
+		{folder: "vsphere_linux_options_and_network_interface", flags: []string{}, exitCode: 0}, //do not warn for known uninstalled plugins components
```

Comment:
So anything using a bundled plugin did break. But with the use of knownPlugins I updated the logic in hcl2_upgrade to no longer report an error if the unknown plugins is a known plugin that will be added to the required_plugins block. The updated logic looks like 

```go
if !p.BuilderPlugins.Has(builderCfg.Type) && knownPluginComponent(builderCfg.Type) == unknownPluginPrefix {
	unknownBuilders = append(unknownBuilders, builderCfg.Type)
}
```


TBH it seems odd to me that we exit with an error but still successfully generate the file. I think we should continue to warn but maybe write that to stderr.

Reply:
Warning for the unknown plugins makes sense, though even if they're known that doesn't mean this will get included to the `required_plugins` block if it's a community one for example.
But I agree, if the template is generated and didn't error in the middle of serialisation, this should probably exit 0.

Reply:
> Warning for the unknown plugins makes sense, though even if they're known that doesn't mean this will get included to the `required_plugins` block if it's a community one for example.
> But I agree, if the template is generated and didn't error in the middle of serialisation, this should probably exit 0.

This will be done in a separate PR. It looks like there are a few errors we are "ignoring" but still exiting with 1. This change is a little out of scope for the plugins so deferring until later.

Reply:
This PR is ready for a second review.